File: WebApplication/src/signalR/notification/dtos/notification.dto.ts
import { ReceivedVoucherState } from "src/received-vouchers/enums/received-voucher-state.enum";
import { VoucherRequestState } from "../../../voucher-requests/enums/voucher-request-state.enum";
import { NotificationEntityType } from "../enums/notification-entity-type.enum";
import { NotificationType } from "../enums/notification-type.enum";

export class NotificationDto {
    entityId: number;

    type: NotificationType;
    entityType: NotificationEntityType;

    companyId: number;
    offeringId: number;
    supplierId: number;

    createDate: Date;

    fromUserid: number;
    fromUsername: string;
    fromFullname: string;
    fromUserOrganization: string;

    toUserId: number;

    text: string;

    // Only if EntityType == NotificationEntityType.VoucherRequest && Type == NotificationType.ChangedState
    vrState: VoucherRequestState;
    // Only if EntityType == NotificationEntityType.VoucherRequest && Type == NotificationType.GeneratedCode
    code: string;

    // Only if EntityType == NotificationEntityType.ReceivedVoucher && Type == NotificationType.ChangedState
    rvState: ReceivedVoucherState;
}

File: Sc.Models/Dtos/Notifications/NotificationDto.cs
using Sc.Models.Enums.Notifications;
using Sc.Models.Enums.ReceivedVouchers;
using Sc.Models.Enums.VoucherRequests;

namespace Sc.Models.Dtos.Notifications
{
    public class NotificationDto
    {
        public int EntityId { get; set; }

        public NotificationType Type { get; set; }
        public NotificationEntityType EntityType { get; set; }

        public int CompanyId { get; set; }
        public int OfferingId { get; set; }
        public int SupplierId { get; set; }

        public DateTime CreateDate { get; set; }

        public int FromUserId { get; set; }
        public string FromUsername { get; set; }
        public string FromFullname { get; set; }
        public string FromUserOrganization { get; set; }

        public int ToUserId { get; set; }

        public string Text { get; set; }

        // Only if EntityType == NotificationEntityType.VoucherRequest && Type == NotificationType.ChangedState
        public VoucherRequestState? VrState { get; set; }
        // Only if EntityType == NotificationEntityType.VoucherRequest && Type == NotificationType.GeneratedCode
        public string Code { get; set; }

        // Only if EntityType == NotificationEntityType.ReceivedVoucher && Type == NotificationType.ChangedState
        public ReceivedVoucherState? RvState { get; set; }
    }
}
